crates/forge/src/init.rs

```diff
@@ -0,0 +1,122 @@
+use anyhow::{bail, Result};
+
+use include_dir::{include_dir, Dir, DirEntry};
+use std::ffi::OsStr;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+static TEMPLATE: Dir = include_dir!("starknet_forge_template");
+
+fn create_with_template(dir: &Dir<'_>, base_path: &PathBuf, project_name: &str) -> Result<()> {
+    fs::create_dir_all(base_path)?;
```

Comment:
Is recursively creating a project dir a safe approach? I think we could assume that the dir from which `snforge` is run already exists anyway so we only should need to create a directory with one level of nesting here.